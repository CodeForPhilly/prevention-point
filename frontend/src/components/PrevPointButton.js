import React from "react"
import PropTypes from "prop-types"
import { makeStyles } from "@material-ui/styles"
import clsx from "clsx"
import Button from "@material-ui/core/Button"

const useStyles = makeStyles(() => ({
  PrevPointButton: {
    marginTop: 20,
  },
}))

const PrevPointButton = props => {
  // can add a class where button is imported
  const classes = useStyles()
  const { children, className } = props

  return (
    <Button
      {...props}
      className={clsx(classes.PrevPointButton, className)}
      variant="contained"
    >
      {children}
    </Button>
  )
}

PrevPointButton.propTypes = {
  type: PropTypes.string,
  color: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.node,
  disabled: PropTypes.bool,
  className: PropTypes.string,
}

PrevPointButton.defaultProps = {
  size: "small",
  type: "button",
  disabled: false,
  color: "primary",
  onClick: () => {},
}

export default PrevPointButton
