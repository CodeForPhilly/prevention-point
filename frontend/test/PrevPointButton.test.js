import { render } from "@testing-library/react"
import { shallow } from "enzyme"
import React from "react"
import PrevPointButton from "../src/components/PrevPointButton"

// Test the following things:
// 1. Button type: submit or button
// 2. Disabled: true or false
// 3. onclick function
describe("<PrevPointButton />", () => {
  const mockOnClick = jest.fn()
  const text = "Search"
  let props

  beforeEach(() => {
    props = {
      size: "small",
      type: "submit",
      disabled: false,
      color: "primary",
      onClick: mockOnClick,
    }
  })
  it("should render a PrevPointButton component", () => {
    const buttonWrapper = render(
      <PrevPointButton {...props}>{text}</PrevPointButton>
    )
    expect(buttonWrapper).toMatchSnapshot()
  })

  it("should render a PrevPointButton component with Search text", () => {
    const buttonWrapper = render(
      <PrevPointButton {...props}>{text}</PrevPointButton>
    )
    const button = buttonWrapper.queryByText("Search")
    expect(button).toHaveTextContent("Search")
  })

  it("Should have a button type of submit", () => {
    const button = shallow(<PrevPointButton {...props}>{text}</PrevPointButton>)
    expect(button.props().type).toBe("submit")
  })

  it("should have disabled set to false", () => {
    const button = shallow(<PrevPointButton {...props}>{text}</PrevPointButton>)
    expect(button.props().disabled).toBe(false)
  })

  it("should have working onclick function", () => {
    const button = shallow(<PrevPointButton {...props}>{text}</PrevPointButton>)
    button.simulate("click")
    expect(mockOnClick).toHaveBeenCalledTimes(1)
  })
})
