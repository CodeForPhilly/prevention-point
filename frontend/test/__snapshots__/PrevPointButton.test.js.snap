// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrevPointButton /> should render a PrevPointButton component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-PrevPointButton-1 MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Search
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-PrevPointButton-1 MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        Search
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
